src/codegen/function.rs

```diff
@@ -50,17 +52,17 @@ pub fn declaration(library: &library::Library, analysis: &analysis::functions::I
     format!("fn {}({}){}", analysis.name, param_str, return_str)
 }
 
-macro_rules! write_to_vec(
-    ($dst:expr, $($arg:tt)*) => (
-        $dst.push(fmt::format(format_args!($($arg)*)))
-    )
-);
+pub fn body(library: &library::Library, analysis: &analysis::functions::Info,
+    in_trait: bool) -> Vec<String> {
+    let mut builder = Builder::new()
+        .glib_name(&analysis.glib_name)
+        .from_glib(analysis.ret.to_return_value_body(&library, &analysis));

```

Comment:
I do not like just 'from_glib' instead `to_return_value_body` for trait.
But for Builder It really means "conversion from glib".
Nothing better not come to mind.